// @jellyfin/sdk/lib/models/api/item-sort-by.js
/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
/**
 * An enum representing the item sort by options.
 * References: https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Model/Querying/ItemSortBy.cs
 * @export
 * @enum {string}
 */
var ItemSortBy;
(function (ItemSortBy) {
    ItemSortBy["AiredEpisodeOrder"] = "AiredEpisodeOrder";
    ItemSortBy["AirTime"] = "AirTime";
    ItemSortBy["Album"] = "Album";
    ItemSortBy["AlbumArtist"] = "AlbumArtist";
    ItemSortBy["Artist"] = "Artist";
    ItemSortBy["CommunityRating"] = "CommunityRating";
    ItemSortBy["CriticRating"] = "CriticRating";
    ItemSortBy["DateCreated"] = "DateCreated";
    ItemSortBy["DateLastContentAdded"] = "DateLastContentAdded";
    ItemSortBy["DatePlayed"] = "DatePlayed";
    ItemSortBy["IsFavoriteOrLiked"] = "IsFavoriteOrLiked";
    ItemSortBy["IsFolder"] = "IsFolder";
    ItemSortBy["IsPlayed"] = "IsPlayed";
    ItemSortBy["IsUnplayed"] = "IsUnplayed";
    ItemSortBy["Name"] = "Name";
    ItemSortBy["OfficialRating"] = "OfficialRating";
    ItemSortBy["PlayCount"] = "PlayCount";
    ItemSortBy["PremiereDate"] = "PremiereDate";
    ItemSortBy["ProductionYear"] = "ProductionYear";
    ItemSortBy["Random"] = "Random";
    ItemSortBy["Runtime"] = "Runtime";
    ItemSortBy["SeriesDatePlayed"] = "SeriesDatePlayed";
    ItemSortBy["SeriesSortName"] = "SeriesSortName";
    ItemSortBy["SortName"] = "SortName";
    ItemSortBy["StartDate"] = "StartDate";
    ItemSortBy["Studio"] = "Studio";
    ItemSortBy["VideoBitRate"] = "VideoBitRate";
})(ItemSortBy || (ItemSortBy = {}));

export { ItemSortBy };
